website/content/en/highlights/2021-07-21-0-16-upgrade-guide.md

```diff
@@ -59,3 +63,100 @@ a measure of the `lua` runtime memory usage, you should update to refer to
 `lua_memory_use_bytes`.
 
 The documentation for this metric has also been updated.
+
+### Vector source/sink version 2 released {#vector_source_sink}
+
+We've released a new major version (`v2`) of our `vector` [source][]/[sink][]
+components. This release resolves several issues and limitations we experienced
+with our previous (`v1`) TCP-based implementation of these two components:
+
+- `vector` sink does not work in k8s with dynamic IP addresses ([#2070][])
+- Allow for HTTP in the vector source and sinks ([#5124][])
+- Allow Vector Source and Sink to Communicate over GRPC ([#6646][])
+- RFC 5843 - Encoding/Decoding for Vector to Vector Communication ([#6032][])
+
+The new version transitions to using gRPC over HTTP as its communication
+protocol, which resolves those limitations.
+
+To allow operators to transition at their leisure, this new release of Vector
+still defaults to `v1`. In the next release (`0.17.0`) we'll require operators
+to explicitly state which version they want to use, but continue to support
+`v1`. The release after that (`0.18.0`) we'll drop `v1` completely, and default
+to `v2`, we also no longer require you to explicitly set the version since there
+will only be one supported going forward.
+
+If you want to opt in to the new (stable!) `v2` version, you can do so as
+follows:
+
+```diff
+[sinks.vector]
+  type = "vector"
++ version = "v2"
+
+[sources.vector]
+  type = "vector"
++ version = "v2"
+```
+
+There are a couple of things to be aware of:
+
+#### Upgrade both the source _and_ sink
+
+You **have** to upgrade **both** the source and sink to `v2`, or none at all,
+you cannot update one without updating the other. Doing so will result in a loss
+of events.
```

Comment:
Technically it only results in event loss if they only upgrade the source, yes?

Reply:
Correct. I don't think it makes sense to mention that, as it's generally _not a good idea_ to mix and match version numbers 😅